File: Task12/Predator.m
//
//  Predator.m
//  Task12
//
//  Created by Sergey Plotnikov on 07.07.16.
//  Copyright © 2016 Sergey Plotnikov. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Predator.h"

@interface Predator()

@property(nonatomic) NSInteger weight;

@end

@implementation Predator

@synthesize name = _name;
@synthesize calories = _calories;
@synthesize weight = _weight;

- (instancetype)initWithName:(NSString *)aName weight:(NSInteger)aWeight
{
    self = [super init];
    
    if (self) {
        _name = aName;
        _calories = kDefaultPredatorCalories;
        _weight = aWeight;
    }
    
    return self;
}

- (void)setCalories:(NSInteger)calories
{
    _calories = calories;
}

@end

File: Task1_2/Predator.swift
//
//  Predator.swift
//  Task1_2
//
//  Created by Admin on 22/09/16.
//  Copyright © 2016 Admin. All rights reserved.
//

import Foundation

class Predator: Animal {

    private (set) var weight: Int
    
    var isProtecting: Bool {
        get {
            return arc4random_uniform(2) == 0
        }
    }
    
    override var description: String {
        get {
            return "Predator " + name + " [cal: \(calories), weight: \(weight)]"
        }
    }

    init(name: String, weight: Int) {
        self.weight = weight
        super.init(name: name, calories: Creature.defalutPredatorCalories)
    }
    
}
